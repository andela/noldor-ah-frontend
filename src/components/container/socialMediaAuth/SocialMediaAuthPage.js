import React, { Component } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import Loading from '../../common/loader/Loader';
import {
  handleSocialMediaAuth
} from '../../../redux/actions/social-media-auth/socialMediaAuthAction';

export class SocialMediaAuthPage extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    const {
      location, handleSocialMedia, provider, history
    } = this.props;
    const btnType = localStorage.getItem('button-type', provider);
    const apiToBecalled = `${location.pathname}/${btnType}/callback${location.search}`;
    handleSocialMedia(apiToBecalled)
      .then(() => history.push('profile'));
  }

  render() {
    const { loading } = this.props;
    if (loading === true) {
      return <Loading />;
    }
    return (
      null
    );
  }
}

SocialMediaAuthPage.propTypes = {
  history: propTypes.object,
  handleSocialMedia: propTypes.func,
  location: propTypes.object,
  provider: propTypes.string,
  loading: propTypes.bool
};

const mapStateToProps = state => ({
  loading: state.socialMediaUserReducer.loading,
  provider: state.socialMediaUserReducer.provider,
});

const mapDispatchToProps = (dispatch) => {
  return ({
    handleSocialMedia(url) {
      return dispatch(handleSocialMediaAuth(url));
    }
  });
};

export default connect(mapStateToProps, mapDispatchToProps)(SocialMediaAuthPage);
