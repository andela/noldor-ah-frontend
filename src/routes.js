import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

const NotFound = () => <div>404</div>;

const Routes = () => (
  <Router>
    <Route exact path="/" component={Home} />
    <Route component={NotFound} />
  </Router>
);

export default Routes;
