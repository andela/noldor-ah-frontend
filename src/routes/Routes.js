import React, { Component } from 'react';
import {
  Router, Route, Switch,
} from 'react-router-dom';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import createHistory from 'history/createBrowserHistory';
import PublicRoute from '../utilities/publicRoute/publicRoute';
import ROUTE_PATH from '../utilities/route-path/routePath';
import Login from '../components/container/login/Login';
import NotFound from '../components/presentational/404/404';
import Signup from '../components/container/signup/Signup';
import { notifyClear } from '../redux/actions/notification/notificationAction';
import EmailVerified from '../components/container/email-verified/EmailVerified';
import Homepage from '../components/container/home/Home';
import Header from '../components/common/header/Header';
import Logout from '../components/container/logout/Logout';
import Footer from '../components/common/footer/Footer';
// eslint-disable-next-line import/no-named-as-default
import SinglePage from '../components/container/article/SingleArticle';
import Profile from '../components/presentational/profile/Profile';

const history = createHistory();

const mapStateToProps = state => ({
  notification: state.notification,
});

export class ConnectedRoutes extends Component {
  constructor(props) {
    super(props);
    const { dispatch } = this.props;

    history.listen(() => {
      dispatch(notifyClear());
    });
  }

  render() {
    return (
      <Router history={history}>
        <div>
          <Header history={history} />
          <Switch>
            <Route path={ROUTE_PATH.homepage} component={Homepage} exact />
            <PublicRoute path={ROUTE_PATH.user.login} component={Login} />
            <Route path={ROUTE_PATH.user.logout} component={Logout} />
            <Route path="/profile" component={Profile} />
            <PublicRoute path={ROUTE_PATH.user.signup} component={Signup} />
            <Route path={ROUTE_PATH.user.verify} component={EmailVerified}/>
            <Route path={ROUTE_PATH.article.single} component={SinglePage} />

            <Route component={NotFound} />
          </Switch>
          <Footer />
        </div>
      </Router>
    );
  }
}

ConnectedRoutes.propTypes = {
  dispatch: propTypes.func.isRequired,
};

const Routes = connect(mapStateToProps)(ConnectedRoutes);

export default Routes;
