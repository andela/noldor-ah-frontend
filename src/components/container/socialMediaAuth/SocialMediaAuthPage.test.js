import React from 'react';
import { shallow } from 'enzyme';
import { SocialMediaAuthPage } from './SocialMediaAuthPage';
import { history } from '../../../routes/Routes';

describe('Visiting the web app through social media registrations or login', () => {
  describe('Test the connected component', () => {
    const handleSocialMedia = jest.fn(() => {
      return Promise.resolve('smurf');
    });
    const location = { pathname: '/auth' };
    const loading = true;

    const wrapper = shallow(
      <SocialMediaAuthPage
        handleSocialMedia={handleSocialMedia}
        location={location}
        history={history}
        loading={loading}
      />
    );
    it('should matche the snapshot', () => {
      const tree = shallow(
        <SocialMediaAuthPage
          handleSocialMedia={handleSocialMedia}
          location={location}
          history={history}/>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should render props', () => {
      expect(wrapper.instance().props.location).toMatchObject({ pathname: '/auth' });
      expect(wrapper.instance().props.loading).toBe(true);
      expect(wrapper.instance().props.history).toMatchObject(history);
    });
  });
});
