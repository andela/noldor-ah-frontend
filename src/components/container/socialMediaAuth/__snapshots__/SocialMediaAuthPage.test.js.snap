// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visiting the web app through social media registrations or login Test the connected component should matche the snapshot 1`] = `""`;
